src/chunks.rs

```diff
@@ -0,0 +1,191 @@
+use anyhow::Context;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Cursor, Seek, SeekFrom};
+use std::path::Path;
+
+fn split(chunk: u64, size: u64) -> Vec<u64> {
+    let e = size / chunk + 1.min(size % chunk);
+    (0..e).map(|i| i * chunk).collect()
+}
+
+#[derive(Debug)]
+pub struct Chunk<C> {
+    lines: std::io::Lines<C>,
+    pos: u64,
+    end: u64,
+}
+
+impl<C> Chunk<C> {
+    pub fn new(
+        mut c: C,
+        chunk: usize,
+        mut current: u64,
+        start: u64,
+        size: u64,
+    ) -> anyhow::Result<(Self, u64)>
+    where
+        C: Seek + BufRead,
```

Comment:
Why do we need to seek if we're memory mapping? We can just get the slice and access at will?

Reply:
Please have a look at the latest version, then it probably makes more sense. I want this to work for every type implements `T: Seek + BufRead` i.e. `BufReader` and `Cursor`. At the moment the actual code uses one the test the other.

Reply:
There's no good reason for the memory not to pass a slice, so this isn't quite the explanation I was looking for.

Reply:
As I've mentioned the implemenation works for files and in memory bytes.